boa3_test/tests/test_interop.py

```diff
@@ -548,76 +548,174 @@ def test_invocation_counter_cant_assign(self):
         self.assertEqual(expected_output, output)
 
     def test_ripemd160_str(self):
-        string = String('test').to_bytes()
         expected_output = (
-            Opcode.PUSHDATA1
-            + Integer(len(string)).to_byte_array(min_length=1)
-            + string
+            Opcode.INITSLOT
+            + b'\x00\x01'
+            + Opcode.LDARG0
             + Opcode.SYSCALL
             + Interop.Ripemd160.interop_method_hash
-            + Opcode.DROP
-            + Opcode.PUSHNULL
             + Opcode.RET
         )
 
         path = '%s/boa3_test/test_sc/interop_test/Ripemd160Str.py' % self.dirname
         output = Boa3.compile(path)
         self.assertEqual(expected_output, output)
 
-    def test_hash160_str(self):
-        string = String('test').to_bytes()
-        expected_output = (
-            Opcode.PUSHDATA1
-            + Integer(len(string)).to_byte_array(min_length=1)
-            + string
-            + Opcode.SYSCALL
-            + Interop.Sha256.interop_method_hash
-            + Opcode.SYSCALL
-            + Interop.Ripemd160.interop_method_hash
-            + Opcode.DROP
-            + Opcode.PUSHNULL
-            + Opcode.RET
-        )
+        import hashlib
+        engine = TestEngine(self.dirname)
+        expected_result = hashlib.new('ripemd160')
```

Comment:
You could have created a variable just for the ripemd160 instead of creating one each time you use it and update it with each value
For example:
```python
ripemd160 = hashlib.new('ripemd160')
ripemd160.update(b'unit test')
expected_result = ripemd160.digest()

...

# don't need to create a new instance with hashlib.new() again
ripemd160.update(b'')
expected_result = ripemd160.digest()
```

Reply:
hash.update() concatenate the bytes, so I'll follow your other tips